src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -221,6 +222,30 @@ namespace ngraph
                 writer << "ctx->mlsl_env->Wait(req);\n";
                 writer.block_end();
             }
+
+            template <>
+            void CPU_Emitter::EMITTER_DECL(ngraph::op::DistBroadcast)
+            {
+                const element::Type& element_type = args[0].get_element_type();
+                auto data_type = "MLSL::DT_FLOAT";
+
+                if (element_type == element::f32)
+                {
+                    data_type = "MLSL::DT_FLOAT";
+                }
+                else if (element_type == element::f64)
+                {
+                    data_type = "MLSL::DT_DOUBLE";
+                }
+
+                writer.block_begin();
+                writer << "MLSL::CommReq* req = ctx->mlsl_dist->DistBroadcast("
```

Comment:
This is different from the one in the builder function `ctx->mlsl_dist->Bcast(arg_tensor, count, data_type, 0, MLSL::GT_DATA)`. Do they both work? Please make sure you are using the same one to avoid any discrepancies.

Also, `MLSL::RT_SUM` is confusing for Bcast.

Reply:
Nice catch on this! Fixed it in the emitter function following the builder function.

Reply:
Thanks! Now this code was tested in the unit-test runs.